packages/gestalt/src/commonTypes.js

```diff
@@ -24,6 +21,9 @@ export type DismissButtonType = {|
   onDismiss: () => void,
 |};
 
+const OnNavigationOptionsPropType: React$PropType$Primitive<OnNavigationOptionsType> =
+  PropTypes.any;
```

Comment:
We definitely want to avoid this if at all possible. Ignoring that Flow error, what would you otherwise type this as?